cpp/src/arrow/filesystem/localfs_benchmark.cc

```diff
@@ -66,7 +66,7 @@ class LocalFSFixture : public benchmark::Fixture {
                                   arrow::internal::PlatformFilename cur_root_dir) {
     ASSERT_OK(arrow::internal::CreateDir(cur_root_dir));
 
-    arrow::internal::StringFormatter<Int32Type> format;
+    arrow::internal::StringFormatter<DoubleType> format;
```

Comment:
Is `DoubleType` correct?  Isn't size_t `Int64Type`?  I'm not actually sure if `size_t`'s width is guaranteed.  So maybe something like...

```
for (size_t i = 0; i < num_files_; ++i) {
...
  format(static_cast<int64_t>(i), ...);
```

Reply:
Ok, here's another suggestion: let's just use `std::stringstream` instead of `StringFormatter`?

The point of having `StringFormatter` is two-fold: 1) performance 2) being locale-agnostic. None of which are concerns here.

Reply:
Since this should be a one-liner change either way, refusing to get a free performance boost by swapping to a `std::stringstream` is somewhat illogical :) I propose we just cast to `int64_t/uint64_t` here.

Reply:
This is benchmark setup code. Furtheremore, I'm highly skeptical that the cost of string formatting would dominate the cost of making system calls to create files and directories...

So let's please keep it simple.

Reply:
oh, yeah I did mean `Int64Type`.

I can also try running with either StringFormatter or stringstream and if the difference isn't significant I can switch to stringstream?

Reply:
oh wait, this is just setup code, I see. then I'll just switch to stringstream.

Reply:
I followed this up with a couple commits to do the following:
1. I changed to Int64Type for completeness, although, I didn't accommodate the cast as recommended.
2. I removed the formatter altogether and the dependency on `formatting.h`

@pitrou I know you recommended using a stringstream, but I realized only 1 append was being done and the string was never re-used, so I just used `operator+` overload and created a string instance directly. I'm not sure if explicitly calling the constructor was necessary, but the aesthetics felt right.

Reply:
also, @ManManson , thanks for the comment!

I verified the benchmarks don't really change (expected from @pitrou 's comment about it being benchmark setup code). Really, there are only 1-4 iterations for each benchmark, so the variance is really high anyways (~10%).